modules/core/src/experimental/utils/gpu-grid-aggregation/gpu-grid-aggregator.js

```diff
@@ -181,155 +227,404 @@ export default class GPUGridAggregator {
     });
   }
 
-  _projectPositions(opts) {
-    let {projectedPositions} = this.state;
-    if (!projectedPositions || opts.changeFlags.dataChanged || opts.changeFlags.viewportChanged) {
-      const {positions, viewport} = opts;
-      projectedPositions = [];
-      for (let index = 0; index < positions.length; index += 2) {
-        const [x, y] = viewport.projectFlat([positions[index], positions[index + 1]]);
-        projectedPositions.push(x, y);
+  /* eslint-disable complexity */
+  _getAggregationParams(opts) {
+    const aggregationParams = Object.assign({}, DEFAULT_RUN_PARAMS, opts);
+    const {
+      useGPU,
+      gridTransformMatrix,
+      viewport,
+      weights,
+      projectPoints,
+      cellSize
+    } = aggregationParams;
+    if (this.state.useGPU !== useGPU) {
+      aggregationParams.changeFlags = DEFAULT_CHANGE_FLAGS;
```

Comment:
You are directly mutating `DEFAULT_CHANGE_FLAGS` here. Should be `Object.assign({}, DEFAULT_CHANGE_FLAGS)`?

Reply:
Good catch, fixed.